cognoml/analysis.py

```diff
@@ -0,0 +1,114 @@
+import collections
+import os
+import warnings
+
+import pandas as pd
+import numpy as np
+import sklearn
+from sklearn.model_selection import train_test_split
+
+from cognoml import utils
+from cognoml.classifiers.logistic_regression import grid_search
+from cognoml.figshare import download_files
+
+# expression_path = os.path.join('download', 'mutation-matrix.tsv.bz2')
+data_directory = "download"
+
+def read_data(version=None):
+    """
+    Read data.
+    """
+    v_dir = download_files(directory=data_directory, artile_id=3487685, version=version)
```

Comment:
`data_directory` is a module scope variable that should be set using:

```python
cognoml.analysis.data_directory = 'new_path'
```

Using the default definition of `directory=data_directory` will break the above functionality.

There is no support for changing `article_id` currently.

So while I see your point, I think the changes add more complexity without any additional functionality at the moment.